compiler.go

```diff
@@ -118,6 +118,16 @@ func (c *Compiler) AddResource(url string, r io.Reader) error {
 	return nil
 }
 
+// AddResourceFromMap adds a schema from an already loaded map to the compiler.
+func (c *Compiler) AddResourceFromMap(url string, doc map[string]any) error {
```

Comment:
`doc` can also be boolean. so I suggest: `AddResourceJSON(url string, doc interface{})`

Reply:
👍 applied!